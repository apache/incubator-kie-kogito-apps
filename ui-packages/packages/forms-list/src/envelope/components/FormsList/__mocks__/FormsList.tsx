import React from 'react';

const MockedFormsList = (): React.ReactElement => {
  return <></>;
};

export default MockedFormsList;
