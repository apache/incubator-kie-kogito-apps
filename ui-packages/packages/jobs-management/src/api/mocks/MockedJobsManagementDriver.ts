import { JobsManagementDriver } from '../JobsManagementDriver';

export const MockedJobsManagementDriver = jest.fn<JobsManagementDriver, []>(
  () => ({
    initialLoad: jest.fn(),
    applyFilter: jest.fn(),
    bulkCancel: jest.fn(),
    cancelJob: jest.fn(),
    rescheduleJob: jest.fn(),
    sortBy: jest.fn(),
    query: jest.fn()
  })
);
