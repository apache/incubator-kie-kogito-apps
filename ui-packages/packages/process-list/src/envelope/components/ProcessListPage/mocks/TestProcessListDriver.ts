/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { ProcessInstance } from '@kogito-apps/management-console-shared';
import {
  ProcessInstanceFilter,
  ProcessListDriver,
  SortBy
} from '../../../../api';

export default class TestProcessListDriver implements ProcessListDriver {
  private readonly processInstances: ProcessInstance[];
  private readonly childProcessInstances: ProcessInstance[];
  private offset: number = 0;
  private limit: number = 10;

  constructor(
    processInstances: ProcessInstance[],
    childProcessInstances: ProcessInstance[]
  ) {
    this.processInstances = processInstances;
    this.childProcessInstances = childProcessInstances;
  }
  /* eslint-disable  @typescript-eslint/no-unused-vars */
  private doSetState(processListFilter: ProcessInstanceFilter, sortBy: SortBy) {
    // do nothing
  }

  initialLoad(filter: ProcessInstanceFilter, sortBy: SortBy): Promise<void> {
    this.doSetState(filter, sortBy);
    return Promise.resolve();
  }

  openProcess(process: ProcessInstance): Promise<void> {
    return Promise.resolve();
  }
  applyFilter(filter: ProcessInstanceFilter): Promise<void> {
    // do nothing
    return Promise.resolve();
  }

  applySorting(sortBy: SortBy): Promise<void> {
    // do nothing
    return Promise.resolve(undefined);
  }

  query(offset: number, limit: number): Promise<ProcessInstance[]> {
    this.offset = offset;
    this.limit = limit;
    return this.doQuery(offset, this.getQueryLimit());
  }

  getChildProcessesQuery(
    rootProcessInstanceId: string
  ): Promise<ProcessInstance[]> {
    return Promise.resolve(this.childProcessInstances.slice(0, 10));
  }
  /* eslint-enable  @typescript-eslint/no-unused-vars */
  private doQuery(start: number, limit: number): Promise<ProcessInstance[]> {
    const queryLimit =
      limit > this.processInstances.length
        ? this.processInstances.length
        : limit;

    return Promise.resolve(this.processInstances.slice(start, queryLimit));
  }

  private getQueryLimit = (): number => {
    return this.offset + this.limit;
  };
}
