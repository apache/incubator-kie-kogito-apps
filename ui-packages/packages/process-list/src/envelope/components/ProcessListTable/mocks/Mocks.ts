/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import {
  ProcessInstance,
  ProcessInstanceState
} from '@kogito-apps/management-console-shared';
export const ProcessInstances: ProcessInstance[] = [
  {
    id: '538f9feb-5a14-4096-b791-2055b38da7c6',
    processId: 'travels',
    businessKey: 'Tra234',
    parentProcessInstanceId: null,
    parentProcessInstance: null,
    processName: 'travels',
    rootProcessInstanceId: null,
    roles: [],
    state: ProcessInstanceState.Error,
    addons: ['jobs-management', 'prometheus-monitoring', 'process-management'],
    start: new Date('2019-10-22T03:40:44.089Z'),
    error: {
      nodeDefinitionId: '__a1e139d5-4e77-48c9-84ae-34578e9817n',
      message: 'Something went wrong'
    },
    lastUpdate: new Date('2019-10-22T03:40:44.089Z'),
    serviceUrl: 'http://localhost:4000',
    endpoint: 'http://localhost:4000',
    variables:
      '{"flight":{"arrival":"2019-10-30T22:00:00Z[UTC]","departure":"2019-10-23T22:00:00Z[UTC]","flightNumber":"MX555"},"trip":{"begin":"2019-10-23T22:00:00Z[UTC]","city":"New York","country":"US","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"hotel":{"address":{"city":"New York","country":"US","street":"street","zipCode":"12345"},"bookingNumber":"XX-012345","name":"Perfect hotel","phone":"09876543"},"traveller":{"address":{"city":"Berlin","country":"Germany","street":"Bakers","zipCode":"100200"},"email":"cristiano@redhat.com","firstName":"Cristiano","lastName":"Nicolai","nationality":"German"}}',
    nodes: [],
    milestones: [],
    childProcessInstances: [
      {
        id: 'c54ca5b0-b975-46e2-a9a0-6a86bf7ac21eajabbcc',
        processId: 'travels',
        businessKey: 'TP444',
        parentProcessInstanceId: '538f9feb-5a14-4096-b791-2055b38da7c6',
        parentProcessInstance: null,
        rootProcessInstanceId: '538f9feb-5a14-4096-b791-2055b38da7c6',
        processName: 'FlightBooking test 2',
        roles: [],
        state: ProcessInstanceState.Active,
        serviceUrl: 'http://localhost:4000',
        lastUpdate: new Date('2019-10-22T03:40:44.089Z'),
        endpoint: 'http://localhost:4000',
        addons: ['process-management'],
        error: {
          nodeDefinitionId: 'a1e139d5-81c77-48c9-84ae-34578e90433n',
          message: 'Something went wrong'
        },
        start: new Date('2019-10-22T03:40:44.089Z'),
        end: new Date('2019-10-22T03:40:44.089Z'),
        variables:
          '{"flight":{"arrival":"2019-10-30T22:00:00Z[UTC]","departure":"2019-10-22T22:00:00Z[UTC]","flightNumber":"MX555"},"trip":{"begin":"2019-10-22T22:00:00Z[UTC]","city":"Berlin","country":"Germany","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"traveller":{"address":{"city":"Karkow","country":"Poland","street":"palna","zipCode":"200300"},"email":"rob@redhat.com","firstName":"Rob","lastName":"Rob","nationality":"Polish"}}',
        nodes: [],
        milestones: [],
        childProcessInstances: []
      }
    ]
  },
  {
    id: '538f9feb-5a14-4096-b791-2055b38da7c6',
    processId: 'travels',
    businessKey: 'Tra234',
    parentProcessInstanceId: null,
    parentProcessInstance: null,
    processName: 'travels',
    rootProcessInstanceId: null,
    roles: [],
    state: ProcessInstanceState.Error,
    addons: ['jobs-management', 'prometheus-monitoring', 'process-management'],
    start: new Date('2019-10-22T03:40:44.089Z'),
    error: {
      nodeDefinitionId: '__a1e139d5-4e77-48c9-84ae-34578e9817n',
      message: 'Something went wrong'
    },
    lastUpdate: new Date('2019-10-22T03:40:44.089Z'),
    serviceUrl: 'http://localhost:4000',
    endpoint: 'http://localhost:4000',
    variables:
      '{"flight":{"arrival":"2019-10-30T22:00:00Z[UTC]","departure":"2019-10-23T22:00:00Z[UTC]","flightNumber":"MX555"},"trip":{"begin":"2019-10-23T22:00:00Z[UTC]","city":"New York","country":"US","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"hotel":{"address":{"city":"New York","country":"US","street":"street","zipCode":"12345"},"bookingNumber":"XX-012345","name":"Perfect hotel","phone":"09876543"},"traveller":{"address":{"city":"Berlin","country":"Germany","street":"Bakers","zipCode":"100200"},"email":"cristiano@redhat.com","firstName":"Cristiano","lastName":"Nicolai","nationality":"German"}}',
    nodes: [],
    milestones: [],
    childProcessInstances: [
      {
        id: 'c54ca5b0-b975-46e2-a9a0-6a86bf7ac21eajabbcc',
        processId: 'travels',
        businessKey: 'TP444',
        parentProcessInstanceId: '538f9feb-5a14-4096-b791-2055b38da7c6',
        parentProcessInstance: null,
        rootProcessInstanceId: '538f9feb-5a14-4096-b791-2055b38da7c6',
        processName: 'FlightBooking test 2',
        roles: [],
        state: ProcessInstanceState.Active,
        serviceUrl: 'http://localhost:4000',
        lastUpdate: new Date('2019-10-22T03:40:44.089Z'),
        endpoint: 'http://localhost:4000',
        addons: ['process-management'],
        error: {
          nodeDefinitionId: 'a1e139d5-81c77-48c9-84ae-34578e90433n',
          message: 'Something went wrong'
        },
        start: new Date('2019-10-22T03:40:44.089Z'),
        end: new Date('2019-10-22T03:40:44.089Z'),
        variables:
          '{"flight":{"arrival":"2019-10-30T22:00:00Z[UTC]","departure":"2019-10-22T22:00:00Z[UTC]","flightNumber":"MX555"},"trip":{"begin":"2019-10-22T22:00:00Z[UTC]","city":"Berlin","country":"Germany","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"traveller":{"address":{"city":"Karkow","country":"Poland","street":"palna","zipCode":"200300"},"email":"rob@redhat.com","firstName":"Rob","lastName":"Rob","nationality":"Polish"}}',
        nodes: [],
        milestones: [],
        childProcessInstances: []
      }
    ]
  }
];
