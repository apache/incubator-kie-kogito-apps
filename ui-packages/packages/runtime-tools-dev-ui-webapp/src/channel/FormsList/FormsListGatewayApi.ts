import { FormFilter, FormInfo } from '@kogito-apps/forms-list';
import { getForms } from '../apis';

/* eslint-disable @typescript-eslint/no-empty-interface */
export interface FormsListGatewayApi {
  getFormFilter(): Promise<FormFilter>;
  applyFilter(formList: FormFilter): Promise<void>;
  getFormsQuery(): Promise<FormInfo[]>;
  openForm: (formData: FormInfo) => Promise<void>;
  onOpenFormListen: (listener: OnOpenFormListener) => UnSubscribeHandler;
}

export interface OnOpenFormListener {
  onOpen: (formData: FormInfo) => void;
}

export interface UnSubscribeHandler {
  unSubscribe: () => void;
}

export class FormsListGatewayApiImpl implements FormsListGatewayApi {
  private _FormFilter: FormFilter = {
    formNames: []
  };
  private readonly listeners: OnOpenFormListener[] = [];

  getFormFilter = (): Promise<FormFilter> => {
    return Promise.resolve(this._FormFilter);
  };

  applyFilter = (formFilter: FormFilter): Promise<void> => {
    this._FormFilter = formFilter;
    return Promise.resolve();
  };

  getFormsQuery(): Promise<FormInfo[]> {
    return getForms(this._FormFilter.formNames);
  }

  openForm = (formData: FormInfo): Promise<void> => {
    this.listeners.forEach((listener) => listener.onOpen(formData));
    return Promise.resolve();
  };

  onOpenFormListen(listener: OnOpenFormListener): UnSubscribeHandler {
    this.listeners.push(listener);

    const unSubscribe = () => {
      const index = this.listeners.indexOf(listener);
      if (index > -1) {
        this.listeners.splice(index, 1);
      }
    };

    return {
      unSubscribe
    };
  }
}
