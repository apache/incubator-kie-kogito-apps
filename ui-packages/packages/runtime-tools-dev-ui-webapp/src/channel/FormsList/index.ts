export { FormsListGatewayApi } from './FormsListGatewayApi';
export * from './FormsListContext';
export { default as FormsListContextProvider } from './FormsListContextProvider';
