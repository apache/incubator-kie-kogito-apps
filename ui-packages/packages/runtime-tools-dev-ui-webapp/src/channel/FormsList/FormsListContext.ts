import React, { useContext } from 'react';
import { FormsListGatewayApi } from './FormsListGatewayApi';

const FormsListContext = React.createContext<FormsListGatewayApi>(null);

export const useFormsListGatewayApi = (): FormsListGatewayApi =>
  useContext<FormsListGatewayApi>(FormsListContext);

export default FormsListContext;
