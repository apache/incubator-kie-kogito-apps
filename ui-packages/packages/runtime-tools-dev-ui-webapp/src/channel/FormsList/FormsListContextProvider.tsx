import React from 'react';
import FormsListContext from './FormsListContext';
import { FormsListGatewayApiImpl } from './FormsListGatewayApi';

interface FormsListContextProviderProps {
  children;
}

const FormsListContextProvider: React.FC<FormsListContextProviderProps> = ({
  children
}) => {
  return (
    <FormsListContext.Provider value={new FormsListGatewayApiImpl()}>
      {children}
    </FormsListContext.Provider>
  );
};

export default FormsListContextProvider;
