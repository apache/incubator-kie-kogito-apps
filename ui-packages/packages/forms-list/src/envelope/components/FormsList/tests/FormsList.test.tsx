import React from 'react';
import { mount } from 'enzyme';
import FormsList from '../FormsList';
import { MockedFormsListDriver } from '../../../tests/mocks/MockedFormsListDriver';
import { ToggleGroupItem } from '@patternfly/react-core';
import { act } from 'react-dom/test-utils';
import wait from 'waait';

jest.mock('../../FormsGallery/FormsGallery');
jest.mock('../../FormsListToolbar/FormsListToolbar');
jest.mock('../../FormsTable/FormsTable');

describe('forms list tests', () => {
  const driver = new MockedFormsListDriver();

  it('envelope not connected to channel', () => {
    const props = {
      isEnvelopeConnectedToChannel: false,
      driver: null
    };
    let wrapper;
    act(() => {
      wrapper = mount(<FormsList {...props} />);
    });
    expect(
      wrapper.find(FormsList).props()['isEnvelopeConnectedToChannel']
    ).toBeFalsy();
  });

  it('render forms list - table', () => {
    const props = {
      isEnvelopeConnectedToChannel: true,
      driver: driver
    };
    let wrapper;
    act(() => {
      wrapper = mount(<FormsList {...props} />);
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('render forms list - gallery', async () => {
    const props = {
      isEnvelopeConnectedToChannel: true,
      driver: driver
    };
    let wrapper;
    // switches to gallery view
    act(() => {
      wrapper = mount(<FormsList {...props} />);
      wrapper
        .find(ToggleGroupItem)
        .at(1)
        .find('button')
        .simulate('click');
    });
    await wait(0);
    act(() => {
      wrapper = wrapper.update();
    });
    expect(wrapper).toMatchSnapshot();

    // switches to table view
    act(() => {
      wrapper
        .find(ToggleGroupItem)
        .at(0)
        .find('button')
        .simulate('click');
    });
    await wait(0);
    act(() => {
      wrapper = wrapper.update();
    });
    expect(wrapper.find('MockedFormsTable').exists()).toBeTruthy();
  });
});
