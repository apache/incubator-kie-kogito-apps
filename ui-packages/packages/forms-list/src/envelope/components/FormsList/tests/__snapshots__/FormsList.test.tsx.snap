// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`forms list tests render forms list - gallery 1`] = `
<FormsList
  driver={MockedFormsListDriver {}}
  isEnvelopeConnectedToChannel={true}
>
  <div
    data-ouia-component-type="forms-list"
    data-ouia-safe={true}
  >
    <Split
      hasGutter={true}
    >
      <div
        className="pf-l-split pf-m-gutter"
      >
        <SplitItem>
          <div
            className="pf-l-split__item"
          >
            <MockedFormsListToolbar
              applyFilter={[Function]}
            />
          </div>
        </SplitItem>
        <SplitItem
          isFilled={true}
        >
          <div
            className="pf-l-split__item pf-m-fill"
          />
        </SplitItem>
        <SplitItem
          style={
            Object {
              "padding": "20px",
            }
          }
        >
          <div
            className="pf-l-split__item"
            style={
              Object {
                "padding": "20px",
              }
            }
          >
            <ToggleGroup
              aria-label="switch view toggle group"
            >
              <div
                aria-label="switch view toggle group"
                className="pf-c-toggle-group"
                role="group"
              >
                <ToggleGroupItem
                  aria-label="table view icon button"
                  buttonId="tableView"
                  icon={
                    <BarsIcon
                      color="currentColor"
                      noVerticalAlign={false}
                      size="sm"
                    />
                  }
                  isSelected={false}
                  onChange={[Function]}
                >
                  <div
                    className="pf-c-toggle-group__item"
                  >
                    <button
                      aria-label="table view icon button"
                      aria-pressed={false}
                      className="pf-c-toggle-group__button"
                      id="tableView"
                      onClick={[Function]}
                    >
                      <ToggleGroupItemElement
                        variant="icon"
                      >
                        <span
                          className="pf-c-toggle-group__icon"
                        >
                          <BarsIcon
                            color="currentColor"
                            noVerticalAlign={false}
                            size="sm"
                          >
                            <svg
                              aria-hidden={true}
                              aria-labelledby={null}
                              fill="currentColor"
                              height="1em"
                              role="img"
                              style={
                                Object {
                                  "verticalAlign": "-0.125em",
                                }
                              }
                              viewBox="0 0 448 512"
                              width="1em"
                            >
                              <path
                                d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                              />
                            </svg>
                          </BarsIcon>
                        </span>
                      </ToggleGroupItemElement>
                    </button>
                  </div>
                </ToggleGroupItem>
                <Divider
                  component="div"
                  isVertical={true}
                  key="0 divider"
                >
                  <div
                    className="pf-c-divider pf-m-vertical"
                    role="separator"
                  />
                </Divider>
                <ToggleGroupItem
                  aria-label="card view icon button"
                  buttonId="cardsView"
                  icon={
                    <ThIcon
                      color="currentColor"
                      noVerticalAlign={false}
                      size="sm"
                    />
                  }
                  isSelected={true}
                  onChange={[Function]}
                >
                  <div
                    className="pf-c-toggle-group__item"
                  >
                    <button
                      aria-label="card view icon button"
                      aria-pressed={true}
                      className="pf-c-toggle-group__button pf-m-selected"
                      id="cardsView"
                      onClick={[Function]}
                    >
                      <ToggleGroupItemElement
                        variant="icon"
                      >
                        <span
                          className="pf-c-toggle-group__icon"
                        >
                          <ThIcon
                            color="currentColor"
                            noVerticalAlign={false}
                            size="sm"
                          >
                            <svg
                              aria-hidden={true}
                              aria-labelledby={null}
                              fill="currentColor"
                              height="1em"
                              role="img"
                              style={
                                Object {
                                  "verticalAlign": "-0.125em",
                                }
                              }
                              viewBox="0 0 512 512"
                              width="1em"
                            >
                              <path
                                d="M149.333 56v80c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24V56c0-13.255 10.745-24 24-24h101.333c13.255 0 24 10.745 24 24zm181.334 240v-80c0-13.255-10.745-24-24-24H205.333c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24h101.333c13.256 0 24.001-10.745 24.001-24zm32-240v80c0 13.255 10.745 24 24 24H488c13.255 0 24-10.745 24-24V56c0-13.255-10.745-24-24-24H386.667c-13.255 0-24 10.745-24 24zm-32 80V56c0-13.255-10.745-24-24-24H205.333c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24h101.333c13.256 0 24.001-10.745 24.001-24zm-205.334 56H24c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24h101.333c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24zM0 376v80c0 13.255 10.745 24 24 24h101.333c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H24c-13.255 0-24 10.745-24 24zm386.667-56H488c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H386.667c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24zm0 160H488c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H386.667c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24zM181.333 376v80c0 13.255 10.745 24 24 24h101.333c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H205.333c-13.255 0-24 10.745-24 24z"
                              />
                            </svg>
                          </ThIcon>
                        </span>
                      </ToggleGroupItemElement>
                    </button>
                  </div>
                </ToggleGroupItem>
              </div>
            </ToggleGroup>
          </div>
        </SplitItem>
      </div>
    </Split>
    <Divider>
      <hr
        className="pf-c-divider"
      />
    </Divider>
    <MockedFormsGallery
      driver={MockedFormsListDriver {}}
      formsData={
        Array [
          Object {
            "lastModified": 2020-07-11T18:30:00.000Z,
            "name": "form1",
            "type": "tsx",
          },
          Object {
            "lastModified": 2020-07-11T18:30:00.000Z,
            "name": "form2",
            "type": "html",
          },
        ]
      }
      isLoading={false}
    />
  </div>
</FormsList>
`;

exports[`forms list tests render forms list - table 1`] = `
<FormsList
  driver={MockedFormsListDriver {}}
  isEnvelopeConnectedToChannel={true}
>
  <div
    data-ouia-component-type="forms-list"
    data-ouia-safe={true}
  >
    <Split
      hasGutter={true}
    >
      <div
        className="pf-l-split pf-m-gutter"
      >
        <SplitItem>
          <div
            className="pf-l-split__item"
          >
            <MockedFormsListToolbar
              applyFilter={[Function]}
            />
          </div>
        </SplitItem>
        <SplitItem
          isFilled={true}
        >
          <div
            className="pf-l-split__item pf-m-fill"
          />
        </SplitItem>
        <SplitItem
          style={
            Object {
              "padding": "20px",
            }
          }
        >
          <div
            className="pf-l-split__item"
            style={
              Object {
                "padding": "20px",
              }
            }
          >
            <ToggleGroup
              aria-label="switch view toggle group"
            >
              <div
                aria-label="switch view toggle group"
                className="pf-c-toggle-group"
                role="group"
              >
                <ToggleGroupItem
                  aria-label="table view icon button"
                  buttonId="tableView"
                  icon={
                    <BarsIcon
                      color="currentColor"
                      noVerticalAlign={false}
                      size="sm"
                    />
                  }
                  isSelected={true}
                  onChange={[Function]}
                >
                  <div
                    className="pf-c-toggle-group__item"
                  >
                    <button
                      aria-label="table view icon button"
                      aria-pressed={true}
                      className="pf-c-toggle-group__button pf-m-selected"
                      id="tableView"
                      onClick={[Function]}
                    >
                      <ToggleGroupItemElement
                        variant="icon"
                      >
                        <span
                          className="pf-c-toggle-group__icon"
                        >
                          <BarsIcon
                            color="currentColor"
                            noVerticalAlign={false}
                            size="sm"
                          >
                            <svg
                              aria-hidden={true}
                              aria-labelledby={null}
                              fill="currentColor"
                              height="1em"
                              role="img"
                              style={
                                Object {
                                  "verticalAlign": "-0.125em",
                                }
                              }
                              viewBox="0 0 448 512"
                              width="1em"
                            >
                              <path
                                d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                              />
                            </svg>
                          </BarsIcon>
                        </span>
                      </ToggleGroupItemElement>
                    </button>
                  </div>
                </ToggleGroupItem>
                <Divider
                  component="div"
                  isVertical={true}
                  key="0 divider"
                >
                  <div
                    className="pf-c-divider pf-m-vertical"
                    role="separator"
                  />
                </Divider>
                <ToggleGroupItem
                  aria-label="card view icon button"
                  buttonId="cardsView"
                  icon={
                    <ThIcon
                      color="currentColor"
                      noVerticalAlign={false}
                      size="sm"
                    />
                  }
                  isSelected={false}
                  onChange={[Function]}
                >
                  <div
                    className="pf-c-toggle-group__item"
                  >
                    <button
                      aria-label="card view icon button"
                      aria-pressed={false}
                      className="pf-c-toggle-group__button"
                      id="cardsView"
                      onClick={[Function]}
                    >
                      <ToggleGroupItemElement
                        variant="icon"
                      >
                        <span
                          className="pf-c-toggle-group__icon"
                        >
                          <ThIcon
                            color="currentColor"
                            noVerticalAlign={false}
                            size="sm"
                          >
                            <svg
                              aria-hidden={true}
                              aria-labelledby={null}
                              fill="currentColor"
                              height="1em"
                              role="img"
                              style={
                                Object {
                                  "verticalAlign": "-0.125em",
                                }
                              }
                              viewBox="0 0 512 512"
                              width="1em"
                            >
                              <path
                                d="M149.333 56v80c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24V56c0-13.255 10.745-24 24-24h101.333c13.255 0 24 10.745 24 24zm181.334 240v-80c0-13.255-10.745-24-24-24H205.333c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24h101.333c13.256 0 24.001-10.745 24.001-24zm32-240v80c0 13.255 10.745 24 24 24H488c13.255 0 24-10.745 24-24V56c0-13.255-10.745-24-24-24H386.667c-13.255 0-24 10.745-24 24zm-32 80V56c0-13.255-10.745-24-24-24H205.333c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24h101.333c13.256 0 24.001-10.745 24.001-24zm-205.334 56H24c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24h101.333c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24zM0 376v80c0 13.255 10.745 24 24 24h101.333c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H24c-13.255 0-24 10.745-24 24zm386.667-56H488c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H386.667c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24zm0 160H488c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H386.667c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24zM181.333 376v80c0 13.255 10.745 24 24 24h101.333c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H205.333c-13.255 0-24 10.745-24 24z"
                              />
                            </svg>
                          </ThIcon>
                        </span>
                      </ToggleGroupItemElement>
                    </button>
                  </div>
                </ToggleGroupItem>
              </div>
            </ToggleGroup>
          </div>
        </SplitItem>
      </div>
    </Split>
    <Divider>
      <hr
        className="pf-c-divider"
      />
    </Divider>
    <MockedFormsTable
      driver={MockedFormsListDriver {}}
      formsData={Array []}
      isLoading={true}
      setFormsData={[Function]}
    />
  </div>
</FormsList>
`;
